external/gtest/CMakeLists.txt

```diff
@@ -28,8 +28,15 @@ CMakeExternalProject(
 
 set(GTEST_INCLUDE_DIRS ${gtest_noros_SOURCE_DIR}/include PARENT_SCOPE)
 if (BUILD_SHARED_LIBS)
-  set(GTEST_LIBRARIES ${gtest_noros_BINARY_DIR}/libgtest${CMAKE_SHARED_LIBRARY_SUFFIX}
-    ${gtest_noros_BINARY_DIR}/libgtest_main${CMAKE_SHARED_LIBRARY_SUFFIX})
+
+  if("${ROS_DISTRO}" STREQUAL "noetic")
+    set(GTEST_LIBRARIES ${gtest_noros_BINARY_DIR}/lib/libgtest${CMAKE_SHARED_LIBRARY_SUFFIX}
+  ${gtest_noros_BINARY_DIR}/lib/libgtest_main${CMAKE_SHARED_LIBRARY_SUFFIX})
+  else()
+    set(GTEST_LIBRARIES ${gtest_noros_BINARY_DIR}/libgtest${CMAKE_SHARED_LIBRARY_SUFFIX}
+  ${gtest_noros_BINARY_DIR}/libgtest_main${CMAKE_SHARED_LIBRARY_SUFFIX})
+  endif()
+
```

Comment:
This is very hacky, but we went around for far too long and couldn't get a better alternative